src/components/input/autosuggest/autosuggest.address.js

```diff
@@ -0,0 +1,374 @@
+import AutosuggestUI from './autosuggest.ui';
+import AddressSetter from './autosuggest.address.setter';
+import AddressError from './autosuggest.address.error';
+import { sanitiseAutosuggestText } from './autosuggest.helpers';
+
+import abortableFetch from './abortable-fetch';
+
+export const classInputContainer = 'autosuggest-input';
+export const classNotEditable = 'js-address-not-editable';
+export const classSearch = 'js-address-input__search';
+export const classInput = 'js-autosuggest-input';
+export const classInputUPRN = 'js-hidden-uprn';
+
+export default class AutosuggestAddress {
+  constructor(context) {
+    this.context = context;
+    this.input = context.querySelector(`.${classInput}`);
+    this.search = context.querySelector(`.${classSearch}`);
+    this.InputUPRN = context.querySelector(`.${classInputUPRN}`);
+    this.lang = document.documentElement.getAttribute('lang').toLowerCase();
+    this.addressReplaceChars = [','];
+    this.sanitisedQuerySplitNumsChars = true;
+    this.form = context.closest('form');
+    this.container = context.querySelector(`.${classInputContainer}`);
+    this.errorMessage = this.container.getAttribute('data-error-message');
+    this.APIDomain = this.container.getAttribute('data-api-domain');
+    // State
+    this.currentQuery = null;
+    this.fetch = null;
+    this.currentResults = [];
+    this.totalResults = 0;
+    this.errored = false;
+    this.isEditable = context.querySelector(`.${classNotEditable}`) ? false : true;
+    this.addressSelected = false;
+    this.groupQuery = '';
+
+    // Temporary fix as runner doesn't use full lang code
+    if (this.lang === 'en') {
+      this.lang = 'en-gb';
+    }
+
+    // Bind event listeners
+    if (this.form) {
+      this.form.addEventListener('submit', this.handleSubmit.bind(this));
+    }
+
+    // Initialise address setter
+    if (this.isEditable) {
+      this.addressSetter = new AddressSetter(context);
+    }
+
+    // Initialise autosuggest
+    this.autosuggest = new AutosuggestUI({
+      context: this.container,
+      onSelect: this.onAddressSelect.bind(this),
+      onUnsetResult: this.addressSetter ? this.addressSetter.onUnsetAddress() : null,
+      lang: this.lang,
+      suggestionFunction: this.suggestAddresses.bind(this),
+      onError: this.onError.bind(this),
+      sanitisedQueryReplaceChars: this.addressReplaceChars,
+      sanitisedQuerySplitNumsChars: this.sanitisedQuerySplitNumsChars,
+      minChars: 3,
+      suggestOnBoot: true,
+      handleUpdate: true,
+    });
+
+    // Set up AIMS api variables and auth
+    this.lookupURL = `${this.APIDomain}/addresses/eq?input=`;
+    this.lookupGroupURL = `${this.APIDomain}/addresses/eq/bucket?`;
+    this.retrieveURL = `${this.APIDomain}/addresses/eq/uprn/`;
+
+    this.user = 'equser';
+    this.password = '$4c@ec1zLBu';
```

Comment:
Is this supposed to be in git? I suppose with whitelisting it doesn't matter?

Reply:
I can't get it to read an env var in runtime so I didn't have any choice.